src/callbacks_step/analysis_dg3d.jl

```diff
@@ -199,6 +199,22 @@ function integrate(func::Func, u,
 end
 
 
+function integrate(func::typeof(enstrophy), u,
+                   mesh::TreeMesh{3},
+                   equations, equations_parabolic::CompressibleNavierStokesDiffusion3D,
+                   dg::DGSEM,
+                   cache, cache_parabolic; normalize=true)
+  gradients_x, gradients_y, gradients_z = cache_parabolic.gradients
+  integrate_via_indices(u, mesh, equations, dg, cache; normalize=normalize) do u, i, j, k, element, equations, dg
+    u_local = get_node_vars(u, equations, dg, i, j, k, element)
+    gradients_1_local = get_node_vars(gradients_x, equations_parabolic, dg, i, j, k, element)
+    gradients_2_local = get_node_vars(gradients_y, equations_parabolic, dg, i, j, k, element)
+    gradients_3_local = get_node_vars(gradients_z, equations_parabolic, dg, i, j, k, element)
+    return func(u_local, (gradients_1_local, gradients_2_local, gradients_3_local), equations_parabolic)
+  end
+end
```

Comment:
Is this specific to `enstrophy` and `CompressibleNavierStokesDiffusion3D` or does it work for similar functions in general? If so, we might generalize it by passing both `equations` and `equations_parabolic` to `func` here.

Reply:
I believe that this structure can work for other integrated quantities that need the gradients. As far as I understand the dispatching there is nothing specific to the enstrophy here.

Reply:
I will try removing theses specifics and see what happens

Reply:
This works, so I have removed the types in d8ecc1a383c36cd621c148b0e3e515b579922db0.

Reply:
Great! What is the motivation to keeping the `::Func` type parameter? Is it because of https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing, or because it is used in the other functions, or is there another reason I don't see?

Reply:
I thought this was necessary to make things faster. To be honest, I just mimicked this structure from the `integrate` function above this new function.

Reply:
OK, then let's keep it like this! /r